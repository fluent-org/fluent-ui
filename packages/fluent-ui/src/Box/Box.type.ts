// https://www.smooth-code.com/open-source/xstyled/docs/system-props/
import * as CSS from 'csstype'
import * as React from 'react'
import * as PropTypes from 'prop-types'
import { StandardProps } from '..'

export interface ResponsiveType<T> {
  xs?: T
  sm?: T
  md?: T
  lg?: T
  xl?: T
}
export type SpaceBaseTypes = number | string | 'auto'
export type SpaceTypes = SpaceBaseTypes | ResponsiveType<SpaceBaseTypes>
export type LayoutTypes = SpaceTypes
export type GridTypes = boolean | 'auto' | ResponsiveType<number>

export interface StylesProps {
  /**
   * Space
   */
  margin?: SpaceTypes
  marginTop?: SpaceTypes
  marginRight?: SpaceTypes
  marginBottom?: SpaceTypes
  marginLeft?: SpaceTypes
  // alias for margin
  m?: SpaceTypes
  // alias for marginTop
  mt?: SpaceTypes
  // alias for marginRight
  mr?: SpaceTypes
  // alias for marginBottom
  mb?: SpaceTypes
  // alias for marginLeft
  ml?: SpaceTypes
  // alias for margin-left and margin-right
  mx?: SpaceTypes
  // alias for margin-top and margin-bottom
  my?: SpaceTypes
  padding?: SpaceTypes
  paddingTop?: SpaceTypes
  paddingRight?: SpaceTypes
  paddingBottop?: SpaceTypes
  paddingLeft?: SpaceTypes
  // alias for padding
  p?: SpaceTypes
  // alias for paddingTop
  pt?: SpaceTypes
  // alias for paddingRight
  pr?: SpaceTypes
  // alias for paddingBottom
  pb?: SpaceTypes
  // alias for paddingLeft
  pl?: SpaceTypes
  // alias for padding-left and padding-right
  px?: SpaceTypes
  // alias for padding-top and padding-bottom
  py?: SpaceTypes
  space?: SpaceTypes
  /**
   * Layout
   */
  display?: LayoutTypes
  width?: LayoutTypes
  height?: LayoutTypes
  maxWidth?: LayoutTypes
  minWidth?: LayoutTypes
  maxHeight?: LayoutTypes
  minHeight?: LayoutTypes
  size?: LayoutTypes
  verticalAlign?: CSS.VerticalAlignProperty<SpaceBaseTypes>
  layout?: CSS.DisplayProperty | LayoutTypes | CSS.VerticalAlignProperty<SpaceBaseTypes>
  /**
   * XGrids
   */
  row?: boolean
  col?: GridTypes
  /**
   * Typography
   */
  fontFamily?: CSS.FontFamilyProperty
  fontSize?:
    | CSS.FontSizeProperty<number | string>
    | ResponsiveType<CSS.FontSizeProperty<number | string>>
  fontWeight?: CSS.FontWeightProperty
  lineHeight?: CSS.LineHeightProperty<number>
  textAlign?: CSS.TextAlignProperty
  letterSpacing?: CSS.LetterSpacingProperty<string | number>
  color?: CSS.ColorProperty
  textTransform?: CSS.TextTransformProperty
  typography?: string
  /**
   * Flexboxes
   */
  alignItems?: CSS.AlignItemsProperty
  alignContent?: CSS.AlignContentProperty
  justifyContent?: CSS.JustifyContentProperty
  justifyItems?: CSS.JustifyItemsProperty
  flexWrap?: CSS.FlexWrapProperty
  flexBasis?: CSS.FlexBasisProperty<number>
  flexDirection?: CSS.FlexDirectionProperty
  flex?: CSS.FlexProperty<number>
  justifySelf?: CSS.JustifySelfProperty
  alignSelf?: CSS.AlignSelfProperty
  order?: number | CSS.Globals
  flexboxes?: string
  /**
   * Grids
   */
  gridGap?: CSS.GridGapProperty<number>
  gridColumnGap?: CSS.GridColumnGapProperty<'normal'>
  gridRowGap?: CSS.GridRowGapProperty<'normal'>
  gridColumn?: CSS.GridColumnProperty
  gridRow?: CSS.GridRowProperty
  gridAutoFlow?: CSS.GridAutoFlowProperty
  gridAutoColumns?: CSS.GridAutoColumnsProperty<number>
  gridAutoRows?: CSS.GridAutoRowsProperty<number>
  gridTemplateColumns?: CSS.GridTemplateColumnsProperty<'none' | number>
  gridTemplateRows?: CSS.GridTemplateRowsProperty<'none' | number>
  gridTemplateAreas?: CSS.GridTemplateAreasProperty
  gridArea?: CSS.GridAreaProperty
  grids?: string
  /**
   * Backgrounds
   */
  background?: CSS.BackgroundProperty<number>
  backgroundColor?: CSS.BackgroundColorProperty
  backgroundImage?: CSS.BackgroundImageProperty
  backgroundSize?: CSS.BackgroundSizeProperty<number>
  backgroundPosition?: CSS.BackgroundPositionProperty<number>
  backgroundRepeat?: CSS.BackgroundRepeatProperty
  backgrounds?: string
  /**
   * Positioning
   */
  position?: CSS.PositionProperty
  zIndex?: CSS.ZIndexProperty
  top?: CSS.TopProperty<number>
  right?: CSS.TopProperty<number>
  bottom?: CSS.TopProperty<number>
  left?: CSS.TopProperty<number>
  positioning?: string
  /**
   * Borders
   */
  border?: CSS.BorderProperty<'inherit'>
  borderTop?: CSS.BorderTopProperty<number>
  borderTopColor?: CSS.BorderTopColorProperty
  borderRight?: CSS.BorderRightProperty<number>
  borderRightColor?: CSS.BorderRightProperty<number>
  borderBottom?: CSS.BorderBottomProperty<number>
  borderBottomColor?: CSS.BorderBottomProperty<number>
  borderLeft?: CSS.BorderLeftProperty<number>
  borderLeftColor?: CSS.BorderLeftProperty<number>
  borderColor?: CSS.BorderColorProperty
  borderWidth?: CSS.BorderWidthProperty<number>
  borderStyle?: CSS.BorderStyleProperty
  borderRadius?: CSS.BorderRadiusProperty<number>
  boxShadow?: CSS.BoxShadowProperty
  borders?: string
  /**
   * Basics
   */
  opacity?: LayoutTypes
  overflow?: CSS.OverflowProperty
  transition?: CSS.TransitionProperty
  basics?: string
}

export interface BoxProps extends StylesProps, StandardProps {
  acrylic?: boolean
  children?: React.ReactNode
  as?: keyof JSX.IntrinsicElements
}

export const BoxPropTypes = {
  acrylic: PropTypes.bool,
  children: PropTypes.node,
  as: PropTypes.oneOf<
    | 'symbol'
    | 'object'
    | 'style'
    | 'title'
    | 'time'
    | 'link'
    | 'menu'
    | 'dialog'
    | 'text'
    | 'p'
    | 'col'
    | 'a'
    | 'abbr'
    | 'address'
    | 'area'
    | 'article'
    | 'aside'
    | 'audio'
    | 'b'
    | 'base'
    | 'bdi'
    | 'bdo'
    | 'big'
    | 'blockquote'
    | 'body'
    | 'br'
    | 'button'
    | 'canvas'
    | 'caption'
    | 'cite'
    | 'code'
    | 'colgroup'
    | 'data'
    | 'datalist'
    | 'dd'
    | 'del'
    | 'details'
    | 'dfn'
    | 'div'
    | 'dl'
    | 'dt'
    | 'em'
    | 'embed'
    | 'fieldset'
    | 'figcaption'
    | 'figure'
    | 'footer'
    | 'form'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'head'
    | 'header'
    | 'hgroup'
    | 'hr'
    | 'html'
    | 'i'
    | 'iframe'
    | 'img'
    | 'input'
    | 'ins'
    | 'kbd'
    | 'keygen'
    | 'label'
    | 'legend'
    | 'li'
    | 'main'
    | 'map'
    | 'mark'
    | 'menuitem'
    | 'meta'
    | 'meter'
    | 'nav'
    | 'noindex'
    | 'noscript'
    | 'ol'
    | 'optgroup'
    | 'option'
    | 'output'
    | 'param'
    | 'picture'
    | 'pre'
    | 'progress'
    | 'q'
    | 'rp'
    | 'rt'
    | 'ruby'
    | 's'
    | 'samp'
    | 'script'
    | 'section'
    | 'select'
    | 'small'
    | 'source'
    | 'span'
    | 'strong'
    | 'sub'
    | 'summary'
    | 'sup'
    | 'table'
    | 'template'
    | 'tbody'
    | 'td'
    | 'textarea'
    | 'tfoot'
    | 'th'
    | 'thead'
    | 'tr'
    | 'track'
    | 'u'
    | 'ul'
    | 'var'
    | 'video'
    | 'wbr'
    | 'webview'
    | 'svg'
    | 'animate'
    | 'animateMotion'
    | 'animateTransform'
    | 'circle'
    | 'clipPath'
    | 'defs'
    | 'desc'
    | 'ellipse'
    | 'feBlend'
    | 'feColorMatrix'
    | 'feComponentTransfer'
    | 'feComposite'
    | 'feConvolveMatrix'
    | 'feDiffuseLighting'
    | 'feDisplacementMap'
    | 'feDistantLight'
    | 'feDropShadow'
    | 'feFlood'
    | 'feFuncA'
    | 'feFuncB'
    | 'feFuncG'
    | 'feFuncR'
    | 'feGaussianBlur'
    | 'feImage'
    | 'feMerge'
    | 'feMergeNode'
    | 'feMorphology'
    | 'feOffset'
    | 'fePointLight'
    | 'feSpecularLighting'
    | 'feSpotLight'
    | 'feTile'
    | 'feTurbulence'
    | 'filter'
    | 'foreignObject'
    | 'g'
    | 'image'
    | 'line'
    | 'linearGradient'
    | 'marker'
    | 'mask'
    | 'metadata'
    | 'mpath'
    | 'path'
    | 'pattern'
    | 'polygon'
    | 'polyline'
    | 'radialGradient'
    | 'rect'
    | 'stop'
    | 'switch'
    | 'textPath'
    | 'tspan'
    | 'use'
    | 'view'
  >([
    'symbol',
    'object',
    'style',
    'title',
    'time',
    'link',
    'menu',
    'dialog',
    'text',
    'p',
    'col',
    'a',
    'abbr',
    'address',
    'area',
    'article',
    'aside',
    'audio',
    'b',
    'base',
    'bdi',
    'bdo',
    'big',
    'blockquote',
    'body',
    'br',
    'button',
    'canvas',
    'caption',
    'cite',
    'code',
    'colgroup',
    'data',
    'datalist',
    'dd',
    'del',
    'details',
    'dfn',
    'div',
    'dl',
    'dt',
    'em',
    'embed',
    'fieldset',
    'figcaption',
    'figure',
    'footer',
    'form',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'head',
    'header',
    'hgroup',
    'hr',
    'html',
    'i',
    'iframe',
    'img',
    'input',
    'ins',
    'kbd',
    'keygen',
    'label',
    'legend',
    'li',
    'main',
    'map',
    'mark',
    'menuitem',
    'meta',
    'meter',
    'nav',
    'noindex',
    'noscript',
    'ol',
    'optgroup',
    'option',
    'output',
    'param',
    'picture',
    'pre',
    'progress',
    'q',
    'rp',
    'rt',
    'ruby',
    's',
    'samp',
    'script',
    'section',
    'select',
    'small',
    'source',
    'span',
    'strong',
    'sub',
    'summary',
    'sup',
    'table',
    'template',
    'tbody',
    'td',
    'textarea',
    'tfoot',
    'th',
    'thead',
    'tr',
    'track',
    'u',
    'ul',
    'var',
    'video',
    'wbr',
    'webview',
    'svg',
    'animate',
    'animateMotion',
    'animateTransform',
    'circle',
    'clipPath',
    'defs',
    'desc',
    'ellipse',
    'feBlend',
    'feColorMatrix',
    'feComponentTransfer',
    'feComposite',
    'feConvolveMatrix',
    'feDiffuseLighting',
    'feDisplacementMap',
    'feDistantLight',
    'feDropShadow',
    'feFlood',
    'feFuncA',
    'feFuncB',
    'feFuncG',
    'feFuncR',
    'feGaussianBlur',
    'feImage',
    'feMerge',
    'feMergeNode',
    'feMorphology',
    'feOffset',
    'fePointLight',
    'feSpecularLighting',
    'feSpotLight',
    'feTile',
    'feTurbulence',
    'filter',
    'foreignObject',
    'g',
    'image',
    'line',
    'linearGradient',
    'marker',
    'mask',
    'metadata',
    'mpath',
    'path',
    'pattern',
    'polygon',
    'polyline',
    'radialGradient',
    'rect',
    'stop',
    'switch',
    'textPath',
    'tspan',
    'use',
    'view'
  ])
}
