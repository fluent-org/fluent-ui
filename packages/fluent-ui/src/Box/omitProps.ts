export const omitProps = [
  'theme',
  'display',
  'overflow',
  'width',
  'height',
  'minHeight',
  'minWidth',
  'maxWidth',
  'maxHeight',
  'fontSize',
  'fontFamily',
  'fontWeight',
  'color',
  'bg',
  'background',
  'backgroundClip',
  'backgroundColor',
  'backgroundImage',
  'backgroundOrigin',
  'backgroundPosition',
  'backgroundRepeat',
  'backgroundSize',
  'backgroundAttachment',
  'border',
  'borderTop',
  'borderRight',
  'borderBottom',
  'borderLeft',
  'borderTopWidth',
  'borderRightWidth',
  'borderBottomWidth',
  'borderLeftWidth',
  'borderTopStyle',
  'borderRightStyle',
  'borderBottomStyle',
  'borderLeftStyle',
  'borderTopColor',
  'borderRightColor',
  'borderBottomColor',
  'borderLeftColor',
  'borderRadius',
  'borderTopLeftRadius',
  'borderTopRightRadius',
  'borderBottomRightRadius',
  'borderBottomLeftRadius',
  'margin',
  'marginLeft',
  'marginRight',
  'marginTop',
  'marginBottom',
  'padding',
  'paddingLeft',
  'paddingRight',
  'paddingTop',
  'paddingBottom',
  'boxShadow',
  'flex',
  'flexBasis',
  'flexDirection',
  'flexFlow',
  'flexGrow',
  'flexShrink',
  'flexWrap',
  'order',
  'alignContent',
  'alignItems',
  'alignSelf',
  'justifyContent',
  'placeContent',
  'rowGap',
  'columnGap',
  'gap',
  'padding',
  'margin',
  'zIndex',
  'textAlign',
  'textShadow',
  'gridGap'
]
