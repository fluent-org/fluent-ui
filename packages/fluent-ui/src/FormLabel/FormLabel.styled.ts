import { Style, Styles } from 'jss'
import { FormLabelClassProps } from './FormLabel.type'

const root: Style = {
  display: 'inline-flex',
  justifyContent: 'center',
  alignItems: 'center'
}

const positionTop: Style = {
  flexDirection: 'column'
}
const positionBottom: Style = {
  flexDirection: 'column-reverse'
}
const positionLeft: Style = {
  flexDirection: 'row'
}
const positionRight: Style = {
  flexDirection: 'row-reverse'
}

export const styles: Styles<FormLabelClassProps> = {
  root,
  positionTop,
  positionBottom,
  positionLeft,
  positionRight
}
