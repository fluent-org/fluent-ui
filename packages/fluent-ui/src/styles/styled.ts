import styled from 'styled-components'
import { th, variant, system, breakpoints } from '@xstyled/system'

export { styled, system, th, variant, breakpoints }

export * from 'styled-components'
export * from '@xstyled/system'
// export * from '@xstyled/prop-types'

export * from './stylize'
