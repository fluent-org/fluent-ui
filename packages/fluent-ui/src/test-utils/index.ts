export * from '@testing-library/react'
// @ts-ignore
export * from './createRender'
