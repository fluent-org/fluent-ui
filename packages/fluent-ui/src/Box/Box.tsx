import * as React from 'react'
import classNames from 'classnames'
import { createUseStyles, withTheme } from '@fluent-ui/styles'
import propTypes from '@styled-system/prop-types'
import { styles } from './Box.styled'
import { BoxClassProps, BoxProps, BoxPropTypes } from './Box.type'
import { omit } from '../utils'
import { Theme } from '../styles'

export const name = 'Box'

const useStyles = createUseStyles<Theme, BoxClassProps>(styles, { name })

const Box: React.FC<BoxProps> = React.forwardRef<HTMLDivElement, BoxProps>(
  (props, ref): React.ReactElement => {
    const {
      as: Component = 'div',
      className: classNameProp,
      id,
      style,
      onClick,
      children,
      acrylic = false,
      ...rest
    } = props
    const classes = useStyles(props)
    const className = classNames(
      classes.root,
      classes.system,
      {
        [classes.acrylic]: acrylic
      },
      classNameProp
    )

    const otherProps = omit(rest, [
      'theme',
      'minHeight',
      'minWidth',
      'maxWidth',
      'maxHeight',
      'bg',
      'background',
      'backgroundClip',
      'backgroundColor',
      'backgroundImage',
      'backgroundOrigin',
      'backgroundPosition',
      'backgroundRepeat',
      'backgroundSize',
      'backgroundAttachment',
      'border',
      'boxShadow',
      'flex',
      'flexBasis',
      'flexDirection',
      'flexFlow',
      'flexGrow',
      'flexShrink',
      'flexWrap',
      'order',
      'alignContent',
      'alignItems',
      'alignSelf',
      'justifyContent',
      'placeContent',
      'rowGap',
      'columnGap',
      'gap',
      'padding',
      'margin'
    ])

    const _props = {
      className,
      ref,
      id,
      style,
      onClick,
      ...otherProps
    }

    return <Component {..._props}>{children}</Component>
  }
)

Box.displayName = `F${name}`

Box.propTypes = {
  ...propTypes.space,
  ...propTypes.color,
  ...propTypes.typography,
  ...propTypes.layout,
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.position,
  ...propTypes.shadow,
  ...propTypes.flexbox,
  ...propTypes.grid,
  ...BoxPropTypes
}

// @ts-ignore
export default withTheme(Box)
