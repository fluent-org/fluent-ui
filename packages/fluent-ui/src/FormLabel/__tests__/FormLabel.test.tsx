import * as React from 'react'
import { getClasses, render } from '../../test-utils'
import '@testing-library/jest-dom/extend-expect'

import FormLabel, { name } from '../FormLabel'
import { styles } from '../FormLabel.styled'
import { FormLabelClassProps } from '../FormLabel.type'
import Box from '../../Box'

const classes = getClasses<FormLabelClassProps>(styles, name)

describe('FormLabel', (): void => {
  const testText = 'Hello World'

  test('should be support ref', (): void => {
    const ref = React.createRef<HTMLDivElement>()
    const { container } = render(
      <FormLabel ref={ref} label={testText}>
        <input type="text" />
      </FormLabel>
    )
    const { current: element } = ref
    expect(element).toEqual(container.firstChild)
  })

  test('should render children', (): void => {
    const { container, getByText, sheets } = render(
      <FormLabel label={testText}>
        <input type="text" />
      </FormLabel>
    )
    expect(getByText(testText)).toBeInTheDocument()
    expect(container.firstChild).toHaveClass(classes.root)
    expect(sheets.toString()).toMatchSnapshot()
  })

  test('should be support position', (): void => {
    const { container, getByTestId } = render(
      <Box>
        <FormLabel label={testText} position="top" data-testid="top">
          <input type="text" />
        </FormLabel>
        <FormLabel label={testText} position="bottom" data-testid="bottom">
          <input type="text" />
        </FormLabel>
        <FormLabel label={testText} position="left" data-testid="left">
          <input type="text" />
        </FormLabel>
        <FormLabel label={testText} position="right" data-testid="right">
          <input type="text" />
        </FormLabel>
      </Box>
    )
    expect(getByTestId('top')).toHaveClass(classes.positionTop)
    expect(getByTestId('bottom')).toHaveClass(classes.positionBottom)
    expect(getByTestId('left')).toHaveClass(classes.positionLeft)
    expect(getByTestId('right')).toHaveClass(classes.positionRight)
    expect(container.firstChild).toMatchSnapshot()
  })
})
