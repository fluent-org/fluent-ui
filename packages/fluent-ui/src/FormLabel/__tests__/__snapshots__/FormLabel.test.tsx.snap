// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel should be support position 1`] = `
<div
  class="FBox-root FBox-system FBox-system-0"
>
  <label
    class="FBox-root FBox-system FBox-system-2 FTypography-root FTypography-variants FTypography-variants-0 FFormLabel-root FFormLabel-positionTop"
    data-testid="top"
  >
    Hello World
    <input
      style="margin-top: 8px;"
      type="text"
    />
  </label>
  <label
    class="FBox-root FBox-system FBox-system-4 FTypography-root FTypography-variants FTypography-variants-1 FFormLabel-root FFormLabel-positionBottom"
    data-testid="bottom"
  >
    Hello World
    <input
      style="margin-bottom: 8px;"
      type="text"
    />
  </label>
  <label
    class="FBox-root FBox-system FBox-system-6 FTypography-root FTypography-variants FTypography-variants-2 FFormLabel-root FFormLabel-positionLeft"
    data-testid="left"
  >
    Hello World
    <input
      style="margin-left: 8px;"
      type="text"
    />
  </label>
  <label
    class="FBox-root FBox-system FBox-system-8 FTypography-root FTypography-variants FTypography-variants-3 FFormLabel-root FFormLabel-positionRight"
    data-testid="right"
  >
    Hello World
    <input
      style="margin-right: 8px;"
      type="text"
    />
  </label>
</div>
`;

exports[`FormLabel should render children 1`] = `
".FBox-root {
  -webkit-overflow-scrolling: touch;
}
@media screen and (min-width: 600px) {
  .FBox-root {  }
  .FBox-root::-webkit-scrollbar {
    width: 8px;
    background: transparent;
  }
  .FBox-root::-webkit-scrollbar-thumb {
    border-radius: 8px;
    background-color: rgba(0, 0, 0, 0.2);
  }
  .FBox-root::-webkit-scrollbar-thumb:hover {
    background-color: rgba(0, 0, 0, 0.4);
  }
}
  .FBox-system {  }
  .FBox-acrylic {  }
  .FBox-system-0 {  }
  .FBox-acrylic-1 {
    overflow: hidden;
    position: relative;
  }
@supports ((backdrop-filter: blur(10px)) or (-webkit-backdrop-filter: blur(10px))) {
  .FBox-acrylic-1 {
    background: none !important;
    backdrop-filter: blur(10px);
  }
  .FBox-acrylic-1::before, .FBox-acrylic-1::after {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    content: '';
    position: absolute;
  }
  .FBox-acrylic-1::before {
    opacity: 0.5;
    z-index: -2;
  }
  .FBox-acrylic-1::after {
    opacity: 0.25;
    z-index: -1;
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAUVBMVEWFhYWDg4N3d3dtbW17e3t1dXWBgYGHh4d5eXlzc3OLi4ubm5uVlZWPj4+NjY19fX2JiYl/f39ra2uRkZGZmZlpaWmXl5dvb29xcXGTk5NnZ2c8TV1mAAAAG3RSTlNAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEAvEOwtAAAFVklEQVR4XpWWB67c2BUFb3g557T/hRo9/WUMZHlgr4Bg8Z4qQgQJlHI4A8SzFVrapvmTF9O7dmYRFZ60YiBhJRCgh1FYhiLAmdvX0CzTOpNE77ME0Zty/nWWzchDtiqrmQDeuv3powQ5ta2eN0FY0InkqDD73lT9c9lEzwUNqgFHs9VQce3TVClFCQrSTfOiYkVJQBmpbq2L6iZavPnAPcoU0dSw0SUTqz/GtrGuXfbyyBniKykOWQWGqwwMA7QiYAxi+IlPdqo+hYHnUt5ZPfnsHJyNiDtnpJyayNBkF6cWoYGAMY92U2hXHF/C1M8uP/ZtYdiuj26UdAdQQSXQErwSOMzt/XWRWAz5GuSBIkwG1H3FabJ2OsUOUhGC6tK4EMtJO0ttC6IBD3kM0ve0tJwMdSfjZo+EEISaeTr9P3wYrGjXqyC1krcKdhMpxEnt5JetoulscpyzhXN5FRpuPHvbeQaKxFAEB6EN+cYN6xD7RYGpXpNndMmZgM5Dcs3YSNFDHUo2LGfZuukSWyUYirJAdYbF3MfqEKmjM+I2EfhA94iG3L7uKrR+GdWD73ydlIB+6hgref1QTlmgmbM3/LeX5GI1Ux1RWpgxpLuZ2+I+IjzZ8wqE4nilvQdkUdfhzI5QDWy+kw5Wgg2pGpeEVeCCA7b85BO3F9DzxB3cdqvBzWcmzbyMiqhzuYqtHRVG2y4x+KOlnyqla8AoWWpuBoYRxzXrfKuILl6SfiWCbjxoZJUaCBj1CjH7GIaDbc9kqBY3W/Rgjda1iqQcOJu2WW+76pZC9QG7M00dffe9hNnseupFL53r8F7YHSwJWUKP2q+k7RdsxyOB11n0xtOvnW4irMMFNV4H0uqwS5ExsmP9AxbDTc9JwgneAT5vTiUSm1E7BSflSt3bfa1tv8Di3R8n3Af7MNWzs49hmauE2wP+ttrq+AsWpFG2awvsuOqbipWHgtuvuaAE+A1Z/7gC9hesnr+7wqCwG8c5yAg3AL1fm8T9AZtp/bbJGwl1pNrE7RuOX7PeMRUERVaPpEs+yqeoSmuOlokqw49pgomjLeh7icHNlG19yjs6XXOMedYm5xH2YxpV2tc0Ro2jJfxC50ApuxGob7lMsxfTbeUv07TyYxpeLucEH1gNd4IKH2LAg5TdVhlCafZvpskfncCfx8pOhJzd76bJWeYFnFciwcYfubRc12Ip/ppIhA1/mSZ/RxjFDrJC5xifFjJpY2Xl5zXdguFqYyTR1zSp1Y9p+tktDYYSNflcxI0iyO4TPBdlRcpeqjK/piF5bklq77VSEaA+z8qmJTFzIWiitbnzR794USKBUaT0NTEsVjZqLaFVqJoPN9ODG70IPbfBHKK+/q/AWR0tJzYHRULOa4MP+W/HfGadZUbfw177G7j/OGbIs8TahLyynl4X4RinF793Oz+BU0saXtUHrVBFT/DnA3ctNPoGbs4hRIjTok8i+algT1lTHi4SxFvONKNrgQFAq2/gFnWMXgwffgYMJpiKYkmW3tTg3ZQ9Jq+f8XN+A5eeUKHWvJWJ2sgJ1Sop+wwhqFVijqWaJhwtD8MNlSBeWNNWTa5Z5kPZw5+LbVT99wqTdx29lMUH4OIG/D86ruKEauBjvH5xy6um/Sfj7ei6UUVk4AIl3MyD4MSSTOFgSwsH/QJWaQ5as7ZcmgBZkzjjU1UrQ74ci1gWBCSGHtuV1H2mhSnO3Wp/3fEV5a+4wz//6qy8JxjZsmxxy5+4w9CDNJY09T072iKG0EnOS0arEYgXqYnXcYHwjTtUNAcMelOd4xpkoqiTYICWFq0JSiPfPDQdnt+4/wuqcXY47QILbgAAAABJRU5ErkJggg==);
  }
}
@supports (not ((backdrop-filter: blur(10px)) or (-webkit-backdrop-filter: blur(10px)))) {
  .FBox-acrylic-1 {
    z-index: 1;
    background: inherit;
  }
  .FBox-acrylic-1::before {
    top: -10px;
    left: -10px;
    width: calc(100% + 20px);
    filter: blur(10px);
    height: calc(100% + 20px);
    content: '';
    z-index: -1;
    position: absolute;
    background: inherit;
  }
  .FBox-acrylic-1::after {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    content: '';
    opacity: 0.5;
    z-index: -1;
    position: absolute;
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAUVBMVEWFhYWDg4N3d3dtbW17e3t1dXWBgYGHh4d5eXlzc3OLi4ubm5uVlZWPj4+NjY19fX2JiYl/f39ra2uRkZGZmZlpaWmXl5dvb29xcXGTk5NnZ2c8TV1mAAAAG3RSTlNAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEAvEOwtAAAFVklEQVR4XpWWB67c2BUFb3g557T/hRo9/WUMZHlgr4Bg8Z4qQgQJlHI4A8SzFVrapvmTF9O7dmYRFZ60YiBhJRCgh1FYhiLAmdvX0CzTOpNE77ME0Zty/nWWzchDtiqrmQDeuv3powQ5ta2eN0FY0InkqDD73lT9c9lEzwUNqgFHs9VQce3TVClFCQrSTfOiYkVJQBmpbq2L6iZavPnAPcoU0dSw0SUTqz/GtrGuXfbyyBniKykOWQWGqwwMA7QiYAxi+IlPdqo+hYHnUt5ZPfnsHJyNiDtnpJyayNBkF6cWoYGAMY92U2hXHF/C1M8uP/ZtYdiuj26UdAdQQSXQErwSOMzt/XWRWAz5GuSBIkwG1H3FabJ2OsUOUhGC6tK4EMtJO0ttC6IBD3kM0ve0tJwMdSfjZo+EEISaeTr9P3wYrGjXqyC1krcKdhMpxEnt5JetoulscpyzhXN5FRpuPHvbeQaKxFAEB6EN+cYN6xD7RYGpXpNndMmZgM5Dcs3YSNFDHUo2LGfZuukSWyUYirJAdYbF3MfqEKmjM+I2EfhA94iG3L7uKrR+GdWD73ydlIB+6hgref1QTlmgmbM3/LeX5GI1Ux1RWpgxpLuZ2+I+IjzZ8wqE4nilvQdkUdfhzI5QDWy+kw5Wgg2pGpeEVeCCA7b85BO3F9DzxB3cdqvBzWcmzbyMiqhzuYqtHRVG2y4x+KOlnyqla8AoWWpuBoYRxzXrfKuILl6SfiWCbjxoZJUaCBj1CjH7GIaDbc9kqBY3W/Rgjda1iqQcOJu2WW+76pZC9QG7M00dffe9hNnseupFL53r8F7YHSwJWUKP2q+k7RdsxyOB11n0xtOvnW4irMMFNV4H0uqwS5ExsmP9AxbDTc9JwgneAT5vTiUSm1E7BSflSt3bfa1tv8Di3R8n3Af7MNWzs49hmauE2wP+ttrq+AsWpFG2awvsuOqbipWHgtuvuaAE+A1Z/7gC9hesnr+7wqCwG8c5yAg3AL1fm8T9AZtp/bbJGwl1pNrE7RuOX7PeMRUERVaPpEs+yqeoSmuOlokqw49pgomjLeh7icHNlG19yjs6XXOMedYm5xH2YxpV2tc0Ro2jJfxC50ApuxGob7lMsxfTbeUv07TyYxpeLucEH1gNd4IKH2LAg5TdVhlCafZvpskfncCfx8pOhJzd76bJWeYFnFciwcYfubRc12Ip/ppIhA1/mSZ/RxjFDrJC5xifFjJpY2Xl5zXdguFqYyTR1zSp1Y9p+tktDYYSNflcxI0iyO4TPBdlRcpeqjK/piF5bklq77VSEaA+z8qmJTFzIWiitbnzR794USKBUaT0NTEsVjZqLaFVqJoPN9ODG70IPbfBHKK+/q/AWR0tJzYHRULOa4MP+W/HfGadZUbfw177G7j/OGbIs8TahLyynl4X4RinF793Oz+BU0saXtUHrVBFT/DnA3ctNPoGbs4hRIjTok8i+algT1lTHi4SxFvONKNrgQFAq2/gFnWMXgwffgYMJpiKYkmW3tTg3ZQ9Jq+f8XN+A5eeUKHWvJWJ2sgJ1Sop+wwhqFVijqWaJhwtD8MNlSBeWNNWTa5Z5kPZw5+LbVT99wqTdx29lMUH4OIG/D86ruKEauBjvH5xy6um/Sfj7ei6UUVk4AIl3MyD4MSSTOFgSwsH/QJWaQ5as7ZcmgBZkzjjU1UrQ74ci1gWBCSGHtuV1H2mhSnO3Wp/3fEV5a+4wz//6qy8JxjZsmxxy5+4w9CDNJY09T072iKG0EnOS0arEYgXqYnXcYHwjTtUNAcMelOd4xpkoqiTYICWFq0JSiPfPDQdnt+4/wuqcXY47QILbgAAAABJRU5ErkJggg==);
  }
}
  .FTypography-root {
    margin: 0;
  }
  .FTypography-variants {  }
  .FTypography-gutterTop {
    margin-top: 0.65em;
  }
  .FTypography-gutterBottom {
    margin-bottom: 0.35em;
  }
  .FTypography-noWrap {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }
  .FTypography-variants-0 {
    font-size: 0.875rem;
    font-family: Segoe UI,-apple-system,sans-serif,Roboto,Helvetica,Arial;
    font-weight: 400;
    line-height: 1.43;
  }
  .FFormLabel-root {
    display: inline-flex;
    align-items: center;
    justify-content: center;
  }
  .FFormLabel-positionTop {
    flex-direction: column;
  }
  .FFormLabel-positionBottom {
    flex-direction: column-reverse;
  }
  .FFormLabel-positionLeft {
    flex-direction: row;
  }
  .FFormLabel-positionRight {
    flex-direction: row-reverse;
  }"
`;
